src/app/api.service.ts

```diff
@@ -0,0 +1,16 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { environment } from 'src/environments/environment';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ApiService {
+  private endpoint = "users";
```

Comment:
We can rewrite this like:
```
private baseUrl;
constructor(private httpClient: HttpClient) { 
   this.endpoint= `${environment.domain}/users`;
 }
```
In this case we will not reuse domain multiple times for every service.